flexmeasures/data/services/sensors.py

```diff
@@ -76,7 +76,8 @@ def _get_sensor_bdfs_by_source_type(
     for source_type in ("demo script", "user", "forecaster", "scheduler", "reporter"):
         bdf = TimedBelief.search(
             sensors=sensor,
-            most_recent_events_only=True,
+            most_recent_only=True,
+            most_recent_beliefs_only=False,
```

Comment:
Why are these the correct search filters? Wouldn't you only need the most recent belief?

Reply:
So `most_recent_beliefs_only=False` is basically overridden by `most_recent_only=True`? Perhaps leaving `most_recent_beliefs_only` out would be less confusing.

Reply:
I stand by the design decision, which is getting to the heart of the matter: I was puzzled by why you are mixing the two most-recent-x approaches and **not** hitting the `ValueError` that was seemingly set up for exactly the case of mixing the two approaches. I understand was it isn't hit, though (because of the `False` value), but it would be much more legible if you simply leave out the `most_recent_beliefs_only=False,` line (which you can, because it is the default value anyway).

Reply:
The function that is being called here is  `TimedBelief.search()` and [that has a default of `most_recent_beliefs_only=True`](https://github.com/FlexMeasures/flexmeasures/blob/main/flexmeasures/data/models/time_series.py#L703).

Reply:
You're correct. I didn't notice the two different `TimedBelief` classes. I've created an attempt to remedy the situation, at least to the point at which this line can get the desired behaviour without needing a confusing set of search filters. See #1655